pgjdbc/src/test/java/org/postgresql/test/jdbc4/LobVarlenaTest.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (c) 2005, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.test.jdbc4;
+
+// import static org.junit.Assert.assertEquals;
+// import static org.junit.Assert.assertFalse;
+// import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import org.postgresql.PGProperty;
+import org.postgresql.jdbc.PgBlobBytea;
+import org.postgresql.jdbc.PgClobText;
+import org.postgresql.jdbc.PgConnection;
+import org.postgresql.test.TestUtil;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+import java.io.StringWriter;
+import java.sql.Blob;
+import java.sql.Clob;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Types;
+import java.util.Arrays;
+import java.util.Properties;
+
+public class LobVarlenaTest {
+
+  private Connection conn;
+
+  @Before
+  public void setUp() throws Exception {
+    Properties props = new Properties();
+    props.setProperty(PGProperty.LOB_VARLENA.getName(),"true");
+    conn = TestUtil.openDB(props);
+    TestUtil.createTable(conn, "testlobvarlena",
+                         "textcol text, byteacol bytea");
+    TestUtil.execute("insert into testlobvarlena values("
+                     + "repeat($$ksjkdjkdshtrb\\000ujy2t8mnnksdf$$,500),"
+                     + "decode(repeat($$ksjkds\\000trbut8mnnksdf$$,600),"
+                     + "'escape'))",
+                     conn);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    TestUtil.dropTable(conn, "testlobvarlena");
+    TestUtil.closeDB(conn);
+  }
+
+  @Test
+  public void testConnClob() throws SQLException {
+    // test we can get a clob from the connection and do basic set, get,
+    // and search operations on it
+    assertTrue(((PgConnection)conn).getLobVarlena());
+    Clob clob = conn.createClob();
+    assertTrue(clob instanceof PgClobText);
+    assertTrue(clob.length() == 0);
+    clob.setString(1,"foobarbaz");
+    assertTrue(clob.length() == 9);
+    assertTrue(clob.toString().equals("foobarbaz"));
+    String s = clob.getSubString(4,3);
+    assertTrue(s.equals("bar"));
+    String s2 = clob.getSubString(7,99);
+    assertTrue(s2.equals("baz"));
+    long pos = clob.position("bar",1);
+    assertTrue(pos == 4);
```

Comment:
Please follow https://github.com/pgjdbc/pgjdbc/blob/master/CONTRIBUTING.md#test

Please keep tests focused, so one can understand what the test does.
Please refrain from `assertTrue` usage as it often produces "expected true got false" errors which are hard to reason about.